engine/src/lib.rs

```diff
@@ -196,7 +202,7 @@ mod contract {
     /// to make any necessary changes to the state such that it aligns with the newly deployed
     /// code.
     #[no_mangle]
-    pub extern "C" fn state_migration() {
+    pub const extern "C" fn state_migration() {
```

Comment:
Similarly here, I think it makes sense to not mark as `const` since a non-trivial implementation would not be `const`.

Reply:
Yes. Makes sense.